### PR TITLE
Feature/api

### PR DESCRIPTION
**Pull request by aCampello**
PR that exposes the prediction as API entrypoint using the service-base from https://github.com/wellcometrust/hal9000/pull/8.

The entrypoint http://localhost:8080 should make predictions on a datapoint or a batch of datapoints:

Screenshot 2021-08-23 at 13 39 59

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Base docker container for tornado service
+FROM 160358319781.dkr.ecr.eu-west-1.amazonaws.com/org.wellcome/ml-services:tornado
```
Comment (jdu) [inline Dockerfile:2]:
```suggestion
FROM 160358319781.dkr.ecr.eu-west-1.amazonaws.com/org.wellcome/ml-services:tornado-latest
```

File: api/entrypoint.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        self.write({"result": result})
+
+        # This is an async handler, so you can carry on and do other things
+        # after having written out the response data to the socket above,
+        # for instance record some metrics with `self.record_metric()`
+        # output some logs or otherwise
+    async def get(self):
+        self.write({
+            "model_name": api_docs["info"]["title"],
+            "version": api_docs["info"]["version"]
```
Comment (jdu) [inline api/entrypoint.py:106]: This should probably output he models_version global instead of what's specified in the docs.

**Review by jdu — CHANGES_REQUESTED** [review]
Only minor things for deployment and consistency

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+RUN pip install -r requirements_minimal.txt
+
+# setup.py needs README.md
+COPY README.md setup.py .
+RUN pip install -e . --no-deps 
+
+COPY api api/
+COPY configs configs/
+
+ENV MODEL_VERSION=2021.07.0
```
[the remark this answers is not in the document]
Comment (aCampello) [reply Dockerfile]: Makes sense. I removed this env variable here: https://github.com/wellcometrust/nutrition-labels/pull/92/commits/a9db29ec0db51a27ada9bf348e38e76651816567 , and added a `--env` flag to the `run-debug` recipe so I can test locally.

File: api/entrypoint.py (hunk 2 above)
Comment (aCampello) [reply api/entrypoint.py]: Makes sense: https://github.com/wellcometrust/nutrition-labels/pull/92/commits/7bfa81ce3d03d748df78e9117aa5f88a10b05352

File: Dockerfile (hunk 1 above)
Comment (aCampello) [reply Dockerfile]: Done!https://github.com/wellcometrust/nutrition-labels/pull/92/commits/3a509619c640be528c6a379a7d5fdc34cdfb7cf5

**Review by jdu — APPROVED** [review]
LGTM! Tested and working in cluster as well.

**Comment by aCampello** [comment]
Merging this! @lizgzil will be glad to know that this is happening :D

**Comment by lizgzil** [comment]
> Merging this! @lizgzil will be glad to know that this is happening :D

this is great! :)

**Pull request merged**